nemo/collections/asr/data/audio_to_diar_label.py

```diff
@@ -0,0 +1,915 @@
+from nemo.core.neural_types import AudioSignal, EncodedRepresentation, LengthsType, NeuralType
+
+
+def get_scale_mapping_list(uniq_timestamps):
+    """
+    Call get_argmin_mat function to find the index of the non-base-scale segment that is closest to the
+    given base-scale segment. For each scale and each segment, a base-scale segment is assigned.
+
+    Args:
+        uniq_timestamps: (Dict)
+            The dictionary containing embeddings, timestamps and multiscale weights.
+            If uniq_timestamps contains only one scale, single scale diarization is performed.
+
+    Returns:
+        scale_mapping_argmat (torch.tensor):
+
+            The element at the m-th row and the n-th column of the scale mapping matrix indicates the (m+1)-th scale
+            segment index which has the closest center distance with (n+1)-th segment in the base scale.
+
+            Example:
+                scale_mapping_argmat[2][101] = 85
+
+            In the above example, it means that 86-th segment in the 3rd scale (python index is 2) is mapped with
+            102-th segment in the base scale. Thus, the longer segments bound to have more repeating numbers since
+            multiple base scale segments (since the base scale has the shortest length) fall into the range of the
+            longer segments. At the same time, each row contains N numbers of indices where N is number of
+            segments in the base-scale (i.e., the finest scale).
+    """
+    uniq_scale_dict = uniq_timestamps['scale_dict']
+    scale_mapping_argmat = [[] for _ in range(len(uniq_scale_dict.keys()))]
+
+    session_scale_mapping_dict = get_argmin_mat(uniq_scale_dict)
+    for scale_idx in sorted(uniq_scale_dict.keys()):
+        scale_mapping_argmat[scale_idx] = torch.tensor(session_scale_mapping_dict[scale_idx])
+    scale_mapping_argmat = torch.stack(scale_mapping_argmat)
+    return scale_mapping_argmat
+
+
+def string_to_float(x, round_digits):
+    """
+    Convert string to float then round the number.
+    """
+    return round(float(x), round_digits)
+
+
+def convert_rttm_line(rttm_line, round_digits=2):
+    """
+    Convert a line in RTTM file to speaker label, start and end timestamps.
+
+    Args:
+        rttm_line (str):
+            A line in RTTM formatted file containing offset and duration of each segment.
+        round_digits (int):
+            Number of digits to be rounded.
+
+    Returns:
+        start (float)
+            Start timestamp in floating point number.
+        end (float):
+            End timestamp in floating point number.
+        speaker (str):
+            speaker string in RTTM lines.
+    """
+    rttm = rttm_line.strip().split()
+    start = string_to_float(rttm[3], round_digits)
+    end = string_to_float(rttm[4], round_digits) + string_to_float(rttm[3], round_digits)
+    speaker = rttm[7]
+    return start, end, speaker
+
+
+def _extract_seg_info_from_rttm(self, uniq_id, rttm_lines, target_spks=None):
```

Comment:
As discussed replace `self` argument here and also in below functions

Reply:
Added.

Reply:
removed self and tested.